[2198] Do not blindly set #threads to requested #threads (4-2-stable)

JustinKyleJames opened the pull request:
Test run of the main suite of tests passed.

trel commented on packaging/resource_suite_s3_nocache.py:
```diff
@@ -2356,6 +2357,42 @@ def test_upload_very_large_file_after_redirect_issues_2104_2114(self):
+                self.user1.assert_icommand('iget -f %s %s' % (filename, get_filename))
+                assert_command(['diff', '-q', filename, get_filename])
+
+                for n_value in [2,3,4,8]:
```
do we want to add a 0 and 1 to this array?

JustinKyleJames replied:
done

trel commented:
if everything passes, please squash, no #.   and we'll wait for another set of eyes.

JustinKyleJames commented:
> if everything passes, please squash, no #. and we'll wait for another set of eyes.

I added an additional test case for a recursive put with -N flag and checksum as that was initially reported as the issue.  It's a little overkill.

I squashed the one additional commit but kept this latest commit so you can take a look at it.

trel commented on packaging/resource_suite_s3_nocache.py:
```diff
@@ -2393,6 +2394,31 @@ def test_iput_with_N_flag_issue_2198(self):
+            # create the local directory
+            os.mkdir(dirname)
+            # create the files
+            lib.make_arbitrary_file('%s/%s'  % (dirname, filename1), 100)
+            lib.make_arbitrary_file('%s/%s'  % (dirname, filename2), 100)
+            lib.make_arbitrary_file('%s/%s'  % (dirname, filename3), 100)
```
double spaces before %

JustinKyleJames replied:
Removed the extra space.

trel commented on packaging/resource_suite_s3_nocache.py:
```diff
@@ -2393,6 +2394,31 @@ def test_iput_with_N_flag_issue_2198(self):
             # clean up the file that was put in case it wasn't cleaned up
             self.user1.assert_icommand('irm -f %s' % filename)
 
+    def recursive_put_small_files_with_checksum_and_N_flag_issue_2198(self):
+
+        dirname = 'recursive_put_small_files_with_checksum_and_N_flag_issue_2198'
```
do we usually use "__issue"... in these names?

JustinKyleJames replied:
Usually.  I made the change.

alanking commented:
Are we planning to do a 4.2.12.2 release?

korydraughn commented:
No due to 4.2.12 being EOL.

JustinKyleJames commented:
> Are we planning to do a 4.2.12.2 release?

I think this is important enough to have a release in 4.2.12.

trel commented:
Agreed.

alanking commented:
Sounds good. I've left a review on the 4-3-stable version. Please let us know if there are any significant differences between the two / whether we should treat this as a distinct review (besides the usual checks).

JustinKyleJames commented:
> Sounds good. I've left a review on the 4-3-stable version. Please let us know if there are any significant differences between the two / whether we should treat this as a distinct review (besides the usual checks).

I will make those changes on this one.  Unfortunately it is a manual process due to the python->python3 update.

I am going to hold off on making the changes for #2204 until that other PR gets approved and merged.

JustinKyleJames commented:
I made and tested the changes from the 4-3-stable PR to this one - issue #2204 changes excluded for now.

korydraughn commented:
What's the current status of this PR?

JustinKyleJames commented:
This pull request is up-to-date.  It is a cherry-pick with necessary alterations for 4.2 - mostly just Python3 -> Python2.

Tests passed.

alanking reviewed (approved):
Compared with https://github.com/irods/irods_resource_plugin_s3/pull/2200 and everything seemed correct to me. # it

JustinKyleJames commented:
> Compared with #2200 and everything seemed correct to me. # it

Done
